pkg/kube/pod.go

```diff
@@ -133,3 +144,69 @@ func WaitForPodCompletion(ctx context.Context, cli kubernetes.Interface, namespa
 	})
 	return errors.Wrap(err, "Pod did not transition into complete state")
 }
+
+// PatchDefaultPodSpecs use Strategic Merge to patch default pod specs with the passed specs
+func PatchDefaultPodSpecs(defaultPodSpecs v1.PodSpec, override map[string]interface{}) (v1.PodSpec, error) {
+	if override == nil {
+		return defaultPodSpecs, nil
+	}
+
+	// Convert override specs to json
+	overrideJson, err := json.Marshal(override)
+	if err != nil {
+		return v1.PodSpec{}, err
+	}
+
+	// Convert default PodSpecs to json
+	originalJson, err := json.Marshal(defaultPodSpecs)
+	if err != nil {
+		return v1.PodSpec{}, err
+	}
+
+	// Merge default specs and override specs with StrategicMergePatch
+	mergedPatch, err := strategicpatch.StrategicMergePatch(originalJson, overrideJson, v1.PodSpec{})
+	if err != nil {
+		return v1.PodSpec{}, err
+	}
+
```

Comment:
Since the code from line 150 to line 170 is repeated in the func `CreateAndMergeJsonPatch ` below, can we add this patch to a separate helper func and call it in both the func to avoid code repetition ?